spot-client/src/spot-remote/analytics/middleware.js

```diff
@@ -76,43 +92,25 @@ MiddlewareRegistry.register(() => next => action => {
         analytics.log(joinCodeEvents.SUBMIT, { shareMode: action.shareMode });
         break;
     }
+    case VIDEO_MUTE: {
+        if (!isPendingAsyncAction(action)) {
+            analytics.log(inCallEvents.VIDEO_MUTE, { muting: action.expectedState });
+        }
+    }
     }
 
     return result;
 });
 
 /**
- * Tracks the {@link REMOTE_CONTROL_REQUEST_STATE} action and logs analytics when the PENDING state
- * is set (it corresponds to the user triggering an action).
+ * Checks if an async action update is notifies that a request is pending.
  *
- * @param {string} requestType - The type of request like audio mute, video mute etc.
- * @param {string} requestState - The request state.
- * @param {any} expectedState - Depends on the request type.
+ * @param {Object} action - The Redux action.
  * @private
- * @returns {void}
+ * @returns {boolean}
  */
-function _remoteControlRequestState({ requestType, requestState, expectedState }) {
-    if (requestState !== requestStates.PENDING) {
-        return;
-    }
-
-    switch (requestType) {
-    case requestTypes.AUDIO_MUTE:
-        analytics.log(inCallEvents.AUDIO_MUTE, { muting: expectedState });
-        break;
-    case requestTypes.SCREENSHARE:
-        if (expectedState === 'proxy') {
-            analytics.log(inCallEvents.WIRELESS_SCREENSHARE_START);
-        } else if (expectedState === 'wired') {
-            analytics.log(inCallEvents.WIRED_SCREENSHARE_START);
-        } else if (expectedState === undefined) {
-            analytics.log(inCallEvents.SCREENSHARE_STOP);
-        }
-        break;
-    case requestTypes.VIDEO_MUTE:
-        analytics.log(inCallEvents.VIDEO_MUTE, { muting: expectedState });
-        break;
-    }
+function isPendingAsyncAction(action) {
+    return action.requestState !== asyncActionRequestStates.PENDING;
```

Comment:
shouldn't it be `action.requestState === asyncActionRequestStates.PENDING` ?

Reply:
:(